Source: Gentoo (Adapted)
Upstream: No
Reason: Fixes compilation with lightdm newer than 1.19.3

diff --git a/src/lightdm-gtk-greeter.c b/src/lightdm-gtk-greeter.c
index 4204d14..80c2a2f 100644
--- a/src/lightdm-gtk-greeter.c
+++ b/src/lightdm-gtk-greeter.c
@@ -948,11 +948,11 @@ start_authentication (const gchar *username)
     {
         gtk_widget_show (GTK_WIDGET (username_entry));
         gtk_widget_show (GTK_WIDGET (cancel_button));
-        lightdm_greeter_authenticate (greeter, NULL);
+        lightdm_greeter_authenticate (greeter, NULL, &error);
     }
     else if (g_strcmp0 (username, "*guest") == 0)
     {
-        lightdm_greeter_authenticate_as_guest (greeter);
+        lightdm_greeter_authenticate_as_guest (greeter, &error);
     }
     else
     {
@@ -972,7 +972,7 @@ start_authentication (const gchar *username)
             set_language (NULL);
         }
 
-        lightdm_greeter_authenticate (greeter, username);
+        lightdm_greeter_authenticate (greeter, username, &error);
     }
 }
 
@@ -982,6 +982,7 @@ cancel_authentication (void)
     GtkTreeModel *model;
     GtkTreeIter iter;
     gboolean other = FALSE;
+    GError *error = NULL;
 
     if (pending_questions)
     {
@@ -994,7 +995,7 @@ cancel_authentication (void)
     if (lightdm_greeter_get_in_authentication (greeter))
     {
         cancelling = TRUE;
-        lightdm_greeter_cancel_authentication (greeter);
+        lightdm_greeter_cancel_authentication (greeter, &error);
         set_message_label ("");
     }
 
@@ -1031,7 +1032,7 @@ start_session (void)
 
     language = get_language ();
     if (language)
-        lightdm_greeter_set_language (greeter, language);
+        lightdm_greeter_set_language (greeter, language, &error);
     g_free (language);
 
     session = get_session ();
@@ -1362,6 +1363,8 @@ G_MODULE_EXPORT
 void
 login_cb (GtkWidget *widget)
 {
+    GError *error = NULL;
+
     /* Reset to default screensaver values */
     if (lightdm_greeter_get_lock_hint (greeter))
         XSetScreenSaver(gdk_x11_display_get_xdisplay(gdk_display_get_default ()), timeout, interval, prefer_blanking, allow_exposures);        
@@ -1375,7 +1378,7 @@ login_cb (GtkWidget *widget)
         start_session ();
     else if (lightdm_greeter_get_in_authentication (greeter))
     {
-        lightdm_greeter_respond (greeter, gtk_entry_get_text (password_entry));
+        lightdm_greeter_respond (greeter, gtk_entry_get_text (password_entry), &error);
         /* If we have questions pending, then we continue processing
          * those, until we are done. (Otherwise, authentication will
          * not complete.) */
